stonesoup/models/measurement/clutter.py

```diff
@@ -0,0 +1,272 @@
+        "It must be a 3x3 matrix. This parameter is only needed when using the normal "
+        "distribution.")
+
+    def function(self, ground_truths: Set[GroundTruthState], position=None, **kwargs) -> Set[Clutter]:
+        """
+        Use the defined distribution and parameters to create simulated clutter
+        for the current time step. Return this clutter to the calling sensor so
+        that it can be added to the measurements.
+
+        Parameters
+        ----------
+        ground_truths : a set of :class:`~.GroundTruthState`
+            The truth states which exist at this time step.
+        position : a 3x1 :class:`~.StateVector` of Cartesian coordinates
+            The position of the sensor which uses this model.
+        
+        Returns
+        -------
+        : set of :class:`~.Clutter`
+            The simulated clutter.
+        """
+
+        # Extract the timestamp from the ground_truths. Groundtruth is
+        # necessary to get the proper timestamp. If there is no
+        # groundtruth return a set of no Clutter.
+        if not ground_truths:
+            return set()
+
+        # Generate the clutter for this time step
+        timestamp = list(ground_truths)[0].timestamp
+        clutter = set()
+        for _ in range(np.random.poisson(self.clutter_rate)):
+            # Call the appropriate helper function according to 
+            # the chosen distribution
+            if self.distribution == 'uniform':
+                random_vector = self.generate_uniform_vector()
+            else:
+                random_vector = self.generate_normal_vector()
+            
+            # Subtract the mounting sensor's position from the clutter,
+            # then convert to spherical coordinates.
+            if position is not None:
+                clutter_vector = random_vector - position[0:3].flatten()
+
+            rho, phi, theta = cart2sphere(*clutter_vector)
+            clutter_vector = np.array([Elevation(theta), Bearing(phi), rho])
+
+            # Create a clutter object. The measurement_model is
+            # inherited from the Detector on which the sensor 
+            # and platform are mounted once it is used.
+            clutter.add(Clutter(state_vector=clutter_vector, 
+                                timestamp=timestamp,
+                                measurement_model=self.measurement_model))
+
+        return clutter
+    
+    def generate_uniform_vector(self):
+        """Generate a random vector in the uniform distribution according
+        to the sensor's defined state space.
+
+        Returns
+        -------
+        : :class:`numpy.array`
+            A uniform random vector with length 3.
+        """
+        x = uniform.rvs(self.state_space[0][0], self.state_space[0][1])
+        y = uniform.rvs(self.state_space[1][0], self.state_space[1][1])
+        z = uniform.rvs(self.state_space[2][0], self.state_space[2][1])
+        return np.array([x, y, z])
```

Comment:
Could probably make parent method more generic, by sampling `n` times from uniform distribution based on size of size of `state_space`.

```python
return np.array([uniform.rvs(*space) for space in self.state_space])
```

Reply:
Yes, great idea. Thanks